python/ql/lib/semmle/python/frameworks/Bottle.qll

```diff
@@ -0,0 +1,179 @@
+/**
+ * Provides classes modeling security-relevant aspects of the `bottle` PyPI package.
+ * See https://bottlepy.org/docs/dev/.
+ */
+
+private import python
+private import semmle.python.Concepts
+private import semmle.python.ApiGraphs
+private import semmle.python.dataflow.new.RemoteFlowSources
+private import semmle.python.frameworks.internal.InstanceTaintStepsHelper
+private import semmle.python.frameworks.Stdlib
+
+/**
+ * INTERNAL: Do not use.
+ *
+ * Provides models for the `bottle` PyPI package.
+ * See https://bottlepy.org/docs/dev/.
+ */
+module Bottle {
+  /** Gets a reference to the `bottle` module. */
+  API::Node bottle() { result = API::moduleImport("bottle") }
+
+  /** Provides models for the `bottle` module. */
+  module BottleModule {
+    /**
+     * Provides models for Bottle applications.
+     */
+    module App {
+      /** Gets class `bottle.Bottle`) */
+      API::Node cls() { result = API::moduleImport("bottle").getMember("Bottle") }
+
+      /** Gets a reference to a Bottle application (an instance of `bottle.Bottle`) */
+      API::Node instance() { result = cls().getReturn() }
+
+      /** Gets a reference to a Bottle application (an instance of `bottle.app`) */
+      API::Node app() { result = bottle().getMember("app").getReturn() }
```

Comment:
I think `app` (like `appstack`) is actually a list that you would have to `pop`, see https://bottlepy.org/docs/dev/api.html#bottle.app and https://bottlepy.org/docs/dev/api.html#bottle.AppStack.pop
```suggestion
      API::Node app() { result = bottle().getMember("Bottle").getReturn() }
```

Reply:
The documentation says "A stack-like list. Calling it returns the head of the stack." so this should be fine.

Reply:
Indeed, that looks ok :-)